### ISSUE TITLE
Handle missing directory

### ISSUE THREAD

**Issue by DaneEveritt** [vote539/socketio-file-upload#58]
Currently if you attempt to upload into a path that does not exist the process simply quits with an error, as would be expected. It'd be nice if there was a way to either have the process create that folder, or to allow an event where I could check for the folder, and create it before the file starts trying to upload to that location.

**Comment by luffs** [comment]
You can create the folder in the handler for the "start" event:
```
var fsExtra = require('fs-extra');
uploader.dir = "/tmp/userFolder";
uploader.on("start", function(event){
    //create the folder if it doesn't exist
    fsExtra.ensureDirSync( uploader.dir );
});
```

**Comment by vote539** [comment]
I made a commit that implements this feature.  See the commit (cc504ed7) and the new section in the docs for the method "pause" for details.  Let me know what you think and then I'll push it live to npm. 👍

**Comment by luffs** [comment]
For me it would be useful if the file metadata was available at the "prestart" event.
The ideal flow for me would be
* Client initiates the upload of a file and sends the metadata
* Server check that the info seems OK, does async-stuff (create a folder / make an database entry)
* Server emits either an accept or abort event
* Client starts sending the actual data

**Comment by luffs** [comment]
I made a pull commit ( https://github.com/vote539/socketio-file-upload/pull/60 ) that enables async validation / preparing. Also fixes memory usage when sending big files, and the progress event.

The usage would be
```
uploader.uploadValidator = function(event, callback){
  fsExtra.ensureDir( uploader.dir, function(err){
    if ( err )
      callback( false ); // abort
    else
      callback( true ); // ready
  });
};
```

**Comment by vote539** [comment]
The pull request #60 has been merged, and this feature is published as v0.6.0.
